Devscord.DiscordFramework/Middlewares/Contexts/UserContext.cs

```diff
@@ -1,45 +1,33 @@
-﻿using Devscord.DiscordFramework.Framework.Architecture.Middlewares;
-using System;
+﻿using Devscord.DiscordFramework.Commons;
+using Devscord.DiscordFramework.Framework.Architecture.Middlewares;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
 
 namespace Devscord.DiscordFramework.Middlewares.Contexts
 {
     public class UserContext : IDiscordContext
     {
         public ulong Id { get; private set; }
         public string Name { get; private set; }
-        public bool IsAdmin { get; private set; }
         public IEnumerable<UserRole> Roles { get; private set; }
         public string AvatarUrl { get; private set; }
         public string Mention { get; private set; }
+        public bool IsAdmin { get; private set; }
+        public bool IsMuted => Roles.Any(x => x.Name == "muted");
```

Comment:
możesz ustawiać to w konstruktorze
nie ma sensu obliczanie tego przy każdym odwołaniu, bo pola są niezmienne (immutable)